src/macaron/dependency_analyzer/java_repo_finder.py

```diff
@@ -150,9 +158,53 @@ def find_scm(pom: Element, tags: list[str]) -> tuple[Iterator[str], int]:
                 # Add the contents of the final tag
                 results.append(element.text.strip())
 
+    # Resolve any Maven properties within the results
+    results = _resolve_properties(pom, results)
+
     return iter(results), len(results)
 
 
+def _resolve_properties(pom: Element, values: list[str]) -> list[str]:
+    """Resolve any Maven properties found within the passed list of values.
+
+    Maven POM files have five different use cases for properties (see https://maven.apache.org/pom.html).
+    Only the two that relate to contents found elsewhere within the same POM file are considered here.
+    That is: ${project.x} where x can be a child tag at any depth, or ${x} where x is found at project.properties.x
+    Entries with rejected properties are removed from the value list.
+    """
+    resolved_values = []
+    print(f"\nValues: {values}\n")
+    for value in values:
+        replacements: list = []
+        # Calculate replacements
+        found = False
+        print(f"\nValue: {value}\n")
+        for match in re.finditer("\\$\\{[^}]+}", value):
+            found = True
+            text = match.group().replace("$", "").replace("{", "").replace("}", "")
+            print(f"Text: {text}")
+            if text.startswith("project."):
+                text = text.replace("project.", "")
+            else:
+                text = f"properties.{text}"
+            print(f"Final Text: {text}")
+            value_iterator, count = find_scm(pom, [text])
+            print(f"Found: {count}")
+            if count == 0:
+                break
+            replacements.insert(0, [match.start(), next(value_iterator), match.end()])
```

Comment:
Why would we want to add to the front of the list (which would be O(n)) instead of appending to the end of the list (which is 0(1))? I think if we want to apply replacements in the reverse order of inserting, we could iterate from index len(replacements) to index 0 :thinking: